src/include/c.h

```diff
@@ -1211,6 +1211,12 @@ typedef union PGAlignedXLogBlock
 	((underlying_type) (expr))
 #endif
 
+#if defined(__clang__) || (defined(__GNUC__) && __GNUC__ >= 7)
+	#define FALL_THROUGH __attribute__((fallthrough));
+#elif
+	#define FALL_THROUGH /* fall through */
```

Comment:
The comment doesn't provide more information than the macro name does. So I suggest deleting it

Reply:
This is not a comment, this is a fallback option

Reply:
When you compile this program using GCC with `-Wimplicit-fallthrough`, the comment fixes the warning.
```c
int main(int argc) {
	int i = 0;
	switch(argc) {
		case 1:
			i+=3;
			/* fall through */
		case 2:
			i+=20;
			/* fall through */
		default:
			return i;
	}
}
```

But in this case preprocessor removes comments before macro substitution. So FALL_THROUGH is replaced with an empty string and the warning is not fixed. It is not possible to replace macro name with comment.
```c
#define FALL_THROUGH /* fall through */

int main(int argc) {
	int i = 0;
	switch(argc) {
		case 1:
			i+=3;
			FALL_THROUGH
		case 2:
			i+=20;
			FALL_THROUGH
		default:
			return i;
	}
}
```

If I'm not right, explain please how can I check that removing of this comment breaks something

Reply:
For GCC and possibly other compilers, comments will be preserved if we do the same thing as in Linux kernel:
```c
#define FALL_THROUGH do {} while (0) /* fallthrough */
```
And use this macro with a semicolon:
```
FALL_THROUGH;
```

But Clang does not recognize fallthrough comments, so the attrubute macros is required for it.

Reply:
If do..while solution works with old compilers, then it's a good solution. I'll check it now

Reply:
We don't use old compilers on CI. We use Ubuntu 22.04 now. And this is only warning, it is not a error. If some old compiler issues a warning, it is not a problem, we can ignore the warning. So I suggest to make the FALL_THROUGH macro empty for other compilers to make the code simplier

Reply:
Modern versions of GCC and Clang are freely available. It is not a problem to install the lastest compiler version

Reply:
So, I changed this macro to support modern compilers only. If a compiler doesn't support __attribute semantics, then it falls back to an empty macro